services/auth/source/ldap/source_sync.go

```diff
@@ -203,19 +204,18 @@ func (source *Source) Sync(ctx context.Context, updateExisting bool) error {
 
 	// Deactivate users not present in LDAP
 	if updateExisting {
-		existPos := 0
-		for i, usr := range users {
-			for existPos < len(existingUsers) && i > existingUsers[existPos] {
-				existPos++
+		for i := range users {
+			usr := users[i]
+			if _, ok := existingUsers[usr.ID]; ok {
+				continue
```

Comment:
The name of `existingUsers` seems a little misleading (existing in database or existing in LDAP?), I read many times to know what it means.

It doesn't mean existing in database nor existing in LDAP, it means "the user can be considered as active in Gitea's database, because it has proper username or email"

I think it needs some comments and maybe a new name is better (eg: "keepActiveUsers")

Or maybe there could be a slice for the "usersToDeactive":

```go
	if usr == nil {
		usersToDeactive = append(usersToDeactive, usr)
	}
...
	if updateExisting {
		for i := range usersToDeactive {
			usr.IsActive = false
			err = user_model.UpdateUserCols(ctx, usr, "is_active")
		}
	}
```

Reply:
`existingUsers` has been renamed to `keepActiveUsers`